### PR TITLE
Bugfix FXIOS-8396 [v124] [Switch Tab Experiment] Duplicated tabs for links that are in Reader Mode

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8396)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18614)

## :bulb: Description

URL for reading mode comes encoded and we need a separate case to check if it's equal with the tab url. Not ideal but the alternative is the check if the url is for reading mode and treat it as a separate case/method. This PR also comes with a fix for the lock icon because localhost urls are treated as not being secure. 

Note: There are some existing bugs for reader mode like webpage being stuck and not reloaded after a while which this PR does not attempt to fix.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by adudenamedruby — APPROVED** [review]
LGTM

**Comment by adudenamedruby** [comment]
Closing for BR.

File: firefox-ios/Client/Frontend/Toolbar+URLBar/TabLocationView.swift [hunk 1]
```diff
@@ -456,7 +457,9 @@ private extension TabLocationView {
         readerModeButton.isHidden = shoppingButton.isHidden ? newReaderModeState == .unavailable : true
         // When the user turns on the reader mode we need to hide the trackingProtectionButton (according to 16400),
         // we will hide it once the newReaderModeState == .active
-        self.trackingProtectionButton.isHidden = newReaderModeState == .active
+        if newReaderModeState == .active {
+            self.trackingProtectionButton.isHidden = true
+        }
```
Comment (lmarceau) [inline firefox-ios/Client/Frontend/Toolbar+URLBar/TabLocationView.swift:462]: Why this change?

Comment (razvanlitianu) [reply firefox-ios/Client/Frontend/Toolbar+URLBar/TabLocationView.swift:462]: When you switch to reader mode, the status is first send as ".available" and for a brief moment you will see the lock icon, then status changes to ".active" and it disappears.
